libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProviderTest.kt

```diff
@@ -0,0 +1,289 @@
+    fun `on fork`() = coroutineRule.runBlockingTest {
+        val primaryRouteGeometryIndex = 200
+        val primaryRouteProgressData = RouteProgressData(4, primaryRouteGeometryIndex, 30)
+        val alternativeMetadata = alternativeRouteMetadata(
+            primaryForkLegIndex = 5,
+            primaryForkRouteGeometryIndex = primaryRouteGeometryIndex,
+            primaryForkLegGeometryIndex = 80,
+            alternativeForkLegIndex = 3,
+            alternativeForkRouteGeometryIndex = 90,
+            alternativeForkLegGeometryIndex = 40,
+            emptyList()
+        )
+        val expected = RouteProgressData(3, 90, 40)
+
+        val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+            primaryRouteProgressData,
+            alternativeMetadata
+        )
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `before fork, alternative starts together with the primary route`() = coroutineRule.runBlockingTest {
+        val primaryRouteProgressData = RouteProgressData(4, 200, 30)
+        val alternativeMetadata = alternativeRouteMetadata(
+            primaryForkLegIndex = 5,
+            primaryForkRouteGeometryIndex = 250,
+            primaryForkLegGeometryIndex = 80,
+            alternativeForkLegIndex = 5,
+            alternativeForkRouteGeometryIndex = 250,
+            alternativeForkLegGeometryIndex = 70,
+            listOf(
+                listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
+                listOf(List(23) { mockk() }),
+                listOf(List(32) { mockk() }, List(19) { mockk() }),
+                listOf(List(14) { mockk() }, List(23) { mockk() }), // 135 points for legs #0-#3
+                listOf(List(8) { mockk() }),
+            )
+        )
+        val expected = RouteProgressData(4, 200, 65)
+
+        val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+            primaryRouteProgressData,
+            alternativeMetadata
+        )
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `before fork, alternative starts after the primary route on the first leg`() =
+        coroutineRule.runBlockingTest {
+            val primaryRouteProgressData = RouteProgressData(4, 200, 35)
+            val alternativeMetadata = alternativeRouteMetadata(
+                primaryForkLegIndex = 5,
+                primaryForkRouteGeometryIndex = 250,
+                primaryForkLegGeometryIndex = 85,
+                alternativeForkLegIndex = 5,
+                alternativeForkRouteGeometryIndex = 220,
+                alternativeForkLegGeometryIndex = 86,
+                listOf(
+                    listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
+                    listOf(List(23) { mockk() }),
+                    listOf(List(32) { mockk() }, List(19) { mockk() }),
+                    listOf(List(14) { mockk() }, List(23) { mockk() }), // 135 points for legs #0-#3
```

Comment:
Shouldn't this say 136 points?
```
(10+6+14)+(22)+(31+18)+(13+22)
```
Not a blocker, I imagine you were thinking about the last index.

Reply:
Well, the last one is duplicated as the starting point in the current leg, so 135 is without this duplicate (I'll update the comment).